packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -1781,8 +1843,13 @@ internal abstract class ManagedDictionaryTester<T>(
             assertFailsWith<RealmException> {
                 values.clear()
             }
-            assertTrue(values.isEmpty())
-            assertTrue(dictionary.isEmpty())
+
+            val iterator = values.iterator()
+            iterator.next()
+            // TODO revisit exception assertion once unified error handling is merged
+            assertFailsWith<RealmException> {
```

Comment:
This should probably be an `llegalStateException` but hopefully the unified error PR will be merged soon.

Reply:
Yeah, as soon as the error handling task is merged all these will fail so it will be easy to fix.